pkg/store/bucket_e2e_test.go

```diff
@@ -433,3 +453,79 @@ func TestBucketStore_ManyParts_e2e(t *testing.T) {
 		testBucketStore_e2e(t, ctx, s)
 	})
 }
+
+func TestBucketStore_TimePartitioning_e2e(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	bkt := inmem.NewBucket()
+
+	dir, err := ioutil.TempDir("", "test_bucket_time_part_e2e")
+	testutil.Ok(t, err)
+	defer func() { testutil.Ok(t, os.RemoveAll(dir)) }()
+
+	series := []labels.Labels{
+		labels.FromStrings("a", "1", "b", "1"),
```

Comment:
Can we reuse `prepareStoreWithTestBlocks`? Series looks exactly the same.

Reply:
I splitted the `prepareTestBlocks` from `prepareStoreWithTestBlocks` so that we can easily configure Store and generate test blocks.

I tried to adjust `prepareStoreWithTestBlocks`, but I didn't like the newly added parameter for filtering and it would need quite a lot of changes due to changes in sync blocks https://github.com/thanos-io/thanos/blob/master/pkg/store/bucket_e2e_test.go#L155-L159

IMO that func is awesome for https://github.com/thanos-io/thanos/blob/master/pkg/store/bucket_e2e_test.go#L365, but doesn't really fit my use case, as I need custom checks in Store gw

Reply:
fair